game-core/src/main/java/games/strategy/engine/framework/ui/background/BackgroundTaskRunner.java

```diff
@@ -11,17 +12,17 @@
 import javax.swing.JFrame;
 import javax.swing.SwingUtilities;
 import javax.swing.SwingWorker;
-import org.triplea.java.Interruptibles;
+import lombok.experimental.UtilityClass;
 import org.triplea.java.function.ThrowingSupplier;
 
 /** Provides methods for running tasks in the background to avoid blocking the UI. */
+@UtilityClass
 public final class BackgroundTaskRunner {
-  private final JFrame frame;
+  private static JFrame mainFrame;
 
-  public BackgroundTaskRunner(final JFrame frame) {
-    checkNotNull(frame);
-
-    this.frame = frame;
+  public static void setMainFrame(final JFrame mainFrame) {
+    Preconditions.checkState(BackgroundTaskRunner.mainFrame == null);
+    BackgroundTaskRunner.mainFrame = mainFrame;
   }
```

Comment:
Hmm I wonder if this really needs to be changed, would anything speak against the callers explicitly passing some sort of frame (not necessarily the main frame) and adding a deprecated getter for the singleton frame if needed?

Reply:
Previously we always passed the main frame, so in part I'm not seeking to change too much at any one time. Passing other frames would be a really different type of change. The getter I'd like to avoid so that we have the 'mainFrame' pushing its reference and using 'setters' rather than other elements using 'getter' to get the reference (essentially an application of IOC, or the "we'll call you, don't call us" principle)